lib/Doctrine/Task.php

```diff
@@ -56,7 +56,7 @@ public function __construct($dispatcher = null)
         $taskName = (string) $this->getTaskName();
 
         //Derive the task name only if it wasn't entered at design-time
-        if (! strlen($taskName)) {
+        if (null === $taskName || ! strlen($taskName)) {
```

Comment:
```suggestion
        if (null === $taskName || 0 !== strlen($taskName)) {
```

Reply:
@thePanz I do not understand this suggestion. It changes the behaviour or the structure ?

Reply:
imho it makes more explicit the strlen evaluation, and the fact that strlen returns an integer (not a boolean)

Reply:
Fixed on 09da1cd899410b2a026fae178ec57440baa62e2a